necromorphs/code/necromorph/abilities/slam.dm

```diff
@@ -0,0 +1,217 @@
+/*
+	Slam is an AOE melee attack, primarily used by the necromorph brute.
+	Slam strikes a three-tile arc either side of an epicentre.
+	Any mobs in the affected area take heavy damage. This damage is multiplied if the victim is lying down, or otherwise incapacitated
+	Any standing mobs in the affected area which are smaller than the attacker, will be knocked down for a time
+	Any dense nonmob atoms in the affected area are hit twice
+*/
+
+/*
+	action/cooldown
+*/
+/datum/action/cooldown/necro/slam
+	name = "Slam"
+	desc = "The brute's signature move."
+	click_to_activate = TRUE
+	cooldown_time = 8 SECOND
+	var/atom/movable/user		//The mob or thing doing the slam attack
+	var/turf/epicentre			//The turf we've targeted with the slam
+	var/damage	=	40			//Base damage dealt
+	var/power = 1				//Used for damage to atoms
+	var/down_factor = 2			//Base damage is multiplied by this on lying targets
+	var/weaken_time = 3			//Number of life ticks victims are weakened (knocked down) for. A life tick is generally 1 second
+	var/windup_time	= 2 SECONDS	//How long the attack is telegraphed for
+
+	//Extra runtime vars
+	var/cached_pixels_x			//Cache the user's pixel offsets so we can revert to them
+	var/cached_pixels_y
+	var/slam_timer				//Timer handle for slam
+	var/x_direction = 0
+	var/list/affected_turfs
+	var/list/affected_turfs_secondary
+
+/datum/action/cooldown/necro/slam/Activate(atom/target)
+	StartCooldown()
+
+	var/direction = get_dir(owner, target)
+	target = get_step(owner, direction)
+
+	//First of all, lets check if we're currently able to charge
+	if (!owner.can_slam(target, TRUE))
+		return
```

Comment:
```suggestion
/datum/action/cooldown/necro/slam/Activate(atom/target)
	// Prevent spamming
	StartCooldown(1 SECONDS)

	var/direction = get_dir(owner, target)
	target = get_step(owner, direction)

	//First of all, lets check if we're currently able to charge
	if (!owner.can_slam(target, TRUE))
		return TRUE
	// Start the real cooldown
	StartCooldown()
```

Reply:
No correct use StartCooldown, because he will display no correct cooldown for players.